tools/client-credentials.sh

```diff
@@ -1,32 +1,34 @@
 #!/bin/bash
 set -e
-## This script package client id and secrets into relevant files
-## The GitHub release includes an install package and a download + install script.
-## Note that a pre-release does not affect the "Latest release" tag, but a regular release does.
-## The release version number argument to this script must match the version number in the settings.gradle
-# file (i.e. version = '0.0.0' line).
+## This script adds client id and secrets into relevant files
 #
 ## Dependencies: jq
 ## Inputs: secretsFile (arg, required) path to the secrets file (valid format)
+##         renderedSecretsFile (arg, required) path to rendered secrets file.
 ##         clientId (arg, required) client id
 ##         clientSecret (arg, required) client secret
-## Usage: ./client-credentials.sh <file> <id> <secret> --> adds client_id and client_secret to file
+## Usage: ./client-credentials.sh <file> <renderedFile> <id> <secret> --> rendered client_id & client_secret to file saving it as renderedFile
 
 secretsFile=$1
-clientId=$2
-clientSecret=$3
+renderedSecretsFile=$2
+clientId=$3
+clientSecret=$4
+
+if [ -z "$secretsFile" ] || [ -z "$renderedSecretsFile" ] || [ "$secretsFile" = "$renderedSecretsFile" ]; then
+  echo "secretsFile & renderedSecretsFile paths are required and must be different"
+  exit 1
+fi
 
 if [ ! -f "$secretsFile" ]; then
-  echo "$secretsFile not available, skipping"
-  exit 0
+  echo "secrets file $secretsFile not available"
+  exit 1
 fi
-tmpFile=$1+".tmp"
 
 if [ -z "$clientId" ] || [ -z "$clientSecret" ]; then
-  echo "client_id & client_secret not available for $secretsFile, skipping"
-  exit 0
+  echo "client_id & client_secret not available for $secretsFile"
+  exit 1
 fi
 
 jq --arg CLIENT_ID "$clientId" --arg CLIENT_SECRET "$clientSecret" \
   '.installed.client_id = $CLIENT_ID | .installed.client_secret = $CLIENT_SECRET' \
-  "$secretsFile" > "$tmpFile" && mv "$tmpFile" "$secretsFile"
+  "$secretsFile" > "$renderedSecretsFile"
```

Comment:
https://google.github.io/styleguide/shellguide.html#variable-expansion

> quote your variables; prefer "${var}" over "$var".

Admittedly, it also says:

> Stay consistent with what you find

but we aren't editing someone else's code, and so should use best practices in code when wherever we can.

Reply:
done, changed